src/quail/sensors/Sensor.cpp

```diff
@@ -27,13 +29,34 @@ void Sensor::activity(){
 
 void Sensor::_configure(Sensor* sensor){
     config_mx.take(NEVER);
-    sensor->configure(); // call the child's implementation of configure()
-    num_sens_cfgd++; // record that another sensor has been configured
-    if(num_sens_cfgd == num_sensors) // check if all sensors have been configured
+    Serial.print("Configured Sensor:");
+    Serial.print(num_sens_cfgd);
+    Serial.print("/");
+    Serial.print(num_sensors);
+    Serial.println("");
+    if(num_sens_cfgd < num_sensors){
+        sensor->configure(); // call the child's implementation of configure()
+        num_sens_cfgd++; // record that another sensor has been configuredred    
+    }
+
+    if (num_sens_cfgd == num_sensors){
+        num_sens_cfgd++;
         sys.tasks.adctask.sensorsConfigured(); // tell ADC task to get goin
+        xEventGroupClearBits(evgroup,ADC_STARTED);
+        xEventGroupSetBits(evgroup,ADC_STARTED_DONE);
+    }
+
     config_mx.give();
 }
 
 void Sensor::ADCbegin(){
+    config_mx.take(NEVER);
+    num_sens_cfgd = 0;
+    Sensor::num_cfgs = 0;
+    PressureSensor::unconfigure();
```

Comment:
instead of hard-coding all the sensor types, we should just loop through all the sensors - it'll call unconfigure() redundantly, but who cares.

Reply:
Yes, but making it in one thread it the end goal.